client-spark/common/src/main/java/org/apache/spark/shuffle/writer/DataPusher.java

```diff
@@ -118,6 +125,18 @@ private synchronized void putBlockId(
         .addAll(blockIds);
   }
 
+  private synchronized void putBlockIdAndShuffle(
```

Comment:
1. Could we give this method a better name? I can't get the point of `put shuffle`.
2. Will it influence the performance of pusher if we use the lock here?

Reply:
> 1. Could we give this method a better name? I can't get the point of `put shuffle`.
> 2. Will it influence the performance of pusher if we use the lock here?

1. The name has been changed.
2. The lock here is the same logic as the putBlockId method above and should not affect it.